crates/uv-distribution/src/metadata/dependency_groups.rs

```diff
@@ -0,0 +1,193 @@
+            if let Some(dev_dependencies) = dev_dependencies {
+                dependency_groups
+                    .entry(DEV_DEPENDENCIES.clone())
+                    .or_insert_with(Vec::new)
+                    .extend(dev_dependencies.clone());
+            }
+
+            dependency_groups
+        };
+
+        // Now that we've resolved the dependency groups, we can validate that each source references
+        // a valid extra or group, if present.
+        Self::validate_sources(project_sources, &dependency_groups)?;
+
+        // Lower the dependency groups.
+        let dependency_groups = dependency_groups
+            .into_iter()
+            .map(|(name, requirements)| {
+                let requirements = match source_strategy {
+                    SourceStrategy::Enabled => requirements
+                        .into_iter()
+                        .flat_map(|requirement| {
+                            let requirement_name = requirement.name.clone();
+                            let group = name.clone();
+                            let extra = None;
+                            LoweredRequirement::from_requirement(
+                                requirement,
+                                project.project_name(),
+                                project.root(),
+                                project_sources,
+                                project_indexes,
+                                extra,
+                                Some(&group),
+                                locations,
+                                project.workspace(),
+                                git_member,
+                            )
+                            .map(
+                                move |requirement| match requirement {
+                                    Ok(requirement) => Ok(requirement.into_inner()),
+                                    Err(err) => Err(MetadataError::GroupLoweringError(
+                                        group.clone(),
+                                        requirement_name.clone(),
+                                        Box::new(err),
+                                    )),
+                                },
+                            )
+                        })
+                        .collect::<Result<Box<_>, _>>(),
+                    SourceStrategy::Disabled => {
+                        Ok(requirements.into_iter().map(Requirement::from).collect())
+                    }
+                }?;
+                Ok::<(GroupName, Box<_>), MetadataError>((name, requirements))
+            })
+            .collect::<Result<BTreeMap<_, _>, _>>()?;
+
+        Ok(Self {
+            name: project.project_name().cloned(),
+            dependency_groups,
+        })
+    }
+
+    /// Validate the sources.
+    ///
+    /// If a source is requested with `group`, ensure that the relevant dependency is
+    /// present in the relevant `dependency-groups` section.
```

Comment:
Maybe silly question, but what happens if the group exists in the workspace root but not in the pyproject.toml we're currently in?